src/coreclr/System.Private.CoreLib/src/System/String.CoreCLR.cs

```diff
@@ -9,6 +10,20 @@ namespace System
 {
     public partial class String
     {
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "String_StrCns")]
+        private static unsafe partial void* StrCnsInternal(uint rid, IntPtr scopeHandle);
+
+        // implementation of CORINFO_HELP_STRCNS
+        [StackTraceHidden]
+        [DebuggerStepThrough]
+        [DebuggerHidden]
+        internal static unsafe string StrCns(uint rid, IntPtr scopeHandle)
+        {
+            void* ptr = StrCnsInternal(rid, scopeHandle);
+            Debug.Assert(ptr != null);
+            return Unsafe.AsRef<string>(ptr);
```

Comment:
```suggestion
            return *(string*)ptr;
```
C# lets you do this directly now.

Reply:
Or you may be even able to change the QCall signature to return `string*`.

Reply:
The pattern existing:  https://github.com/dotnet/runtime/blob/289c443ea97de73b1b485bcaf9dc42e46ed8375a/src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs#L946